src/main/scala/play/api/libs/mailer/MailerPlugin.scala

```diff
@@ -109,6 +112,14 @@ class SMTPMailer @Inject() (smtpConfiguration: SMTPConfiguration) extends Mailer
   }
 }
 
+class ConfigurableMailer @Inject()(smtpConfigurationProvider: Provider[SMTPConfiguration]) extends MailerClient {
```

Comment:
`SMTPMailer` is configurable *once* so `ConfigurableMailer` may be confusing.
Also we should add `SMTP` in the name to make it clear that it will instanciate an `SMTPMailer`, what about `SMTPReconfigurableMailer` or `SMTPReloadableMailer` ?